### ISSUE TITLE
Type instability when multiplying (and/or broadcasting) an SVector of SVectors with a scalar in Julia 0.7 (but not in 0.6)

### ISSUE THREAD

**Issue by Sleort** [JuliaArrays/StaticArrays.jl#480]
Note the type of `0.1b` and `0.1 .* b` in
```
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: https://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.6.4 (2018-07-09 19:09 UTC)
 _/ |\__'_|_|_|\__'_|  |  Official http://julialang.org/ release
|__/                   |  x86_64-pc-linux-gnu

julia> using StaticArrays

julia> a = SVector(1,1); b = SVector(a,a)
2-element StaticArrays.SArray{Tuple{2},StaticArrays.SArray{Tuple{2},Int64,1,2},1,2}:
 [1, 1]
 [1, 1]

julia> 0.1b
2-element StaticArrays.SArray{Tuple{2},StaticArrays.SArray{Tuple{2},Float64,1,2},1,2}:
 [0.1, 0.1]
 [0.1, 0.1]

julia> 0.1.*b
2-element StaticArrays.SArray{Tuple{2},StaticArrays.SArray{Tuple{2},Float64,1,2},1,2}:
 [0.1, 0.1]
 [0.1, 0.1]
```
compared to
```
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: https://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.7.0 (2018-08-08 06:46 UTC)
 _/ |\__'_|_|_|\__'_|  |  Official http://julialang.org/ release
|__/                   |  x86_64-pc-linux-gnu

julia> using StaticArrays

julia> a = SVector(1,1); b = SVector(a,a)
2-element SArray{Tuple{2},SArray{Tuple{2},Int64,1,2},1,2}:
 [1, 1]
 [1, 1]

julia> 0.1b
2-element SArray{Tuple{2},Any,1,2}:
 [0.1, 0.1]
 [0.1, 0.1]

julia> 0.1 .* b
2-element SArray{Tuple{2},Any,1,2}:
 [0.1, 0.1]
 [0.1, 0.1]
```

This must be a bug, right?

**Comment by ryanelandt** [comment]
This doesn't seem to affect Julia 1.0.1 with StaticArrays 0.8.3+.

**Comment by Sleort** [comment]
The issue seems to be fixed in StaticArrays 0.9.
